### PR TITLE
UI: Fix job detail crash when recommendations off

### PR DESCRIPTION
**Pull request by backspace**
Without this, visiting any job detail page on Nomad OSS would
crash with an error like this:

```
Error: Ember Data Request GET /v1/recommendations?job=ping%F0%9F%A5%B3&namespace=default returned a 404
Payload (text/xml)
```

The problem was twofold.

1. The recommendation ability didn’t include anything about checking whether the feature was present. This adds a request to `/v1/operator/license` on application load to determine which features are present and store them in the system service. The ability now looks for `Dynamic Application Sizing` in that feature list.

1. Second, I didn’t check permissions at all in the job-fetching or job detail templates.

Thanks to @DingoEatingFuzz for catching this! 😯

### CODE REVIEW

**Comment by backspace** [comment]
oops wait I only ran the one new test instead of the others which are now broken due to needing the management token 😬

**Comment by backspace** [comment]
okay, all is well now, sorry for the premature review request 😯

File: ui/app/templates/components/job-page/service.hbs [hunk 1]
```diff
@@ -13,9 +13,11 @@
...
   </div>
 
-  {{#each this.job.recommendationSummaries as |summary|}}
-    <Das::RecommendationAccordion @summary={{summary}} />
-  {{/each}}
+  {{#if (can "read agent")}}
+    {{#each this.job.recommendationSummaries as |summary|}}
+      <Das::RecommendationAccordion @summary={{summary}} />
+    {{/each}}
+  {{/if}}
```
Comment (DingoEatingFuzz) [inline ui/app/templates/components/job-page/service.hbs:20]: `read agent` is required here? That's surprising to me. Why is that?

File: ui/app/abilities/recommendation.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { computed } from '@ember/object';
 import { or } from '@ember/object/computed';
 
 export default class Recommendation extends AbstractAbility {
-  @or('bypassAuthorization', 'selfTokenIsManagement', 'policiesSupportAcceptingOnAnyNamespace')
+  @or('selfTokenIsManagement', 'policiesSupportAcceptingOnAnyNamespace')
```
Comment (DingoEatingFuzz) [inline ui/app/abilities/recommendation.js:6]: Can you not use recommendations without ACLs?

**Review by DingoEatingFuzz — COMMENTED** [review]
I have a couple questions regarding the underlying permissions.

File: ui/app/templates/components/job-page/service.hbs (hunk 1 above)
Comment (backspace) [reply ui/app/templates/components/job-page/service.hbs]: ughhhhhhhhh just wrong, fixed in ee0e00d 😞

File: ui/app/abilities/recommendation.js (hunk 2 above)
Comment (backspace) [reply ui/app/abilities/recommendation.js]: hmm… you’re right that this is not a correct way to determine this! 😢

I’ve now added 24c3a95 which introduces a license-fetching step when loading the application akin to when the policies are fetched. On enterprise `GET /v1/operator/license` returns something that we can use to check for whether a feature exists:

```
{
  …,
  License: {
    …,
    Features: […, 'Dynamic Application Sizing', …],
    …,
  …,
}
```

On OSS the endpoint `501`s so I return an empty license with no features instead.

Comment (DingoEatingFuzz) [reply ui/app/abilities/recommendation.js]: 🤩🤩🤩🤩

This is fantastic. I'm sorry this bug ended up being bigger than it appeared, but having this level of feature-checking is an excellent addition.

Comment (backspace) [reply ui/app/abilities/recommendation.js]: I’m glad to have finally added this despite the sadness of another spurious failure message:

image

**Review by DingoEatingFuzz — APPROVED** [review]
Excellent work!

File: ui/app/abilities/recommendation.js (hunk 2 above)
Comment (backspace) [reply ui/app/abilities/recommendation.js]: 😳

**Pull request merged**